### ISSUE TITLE
No HW decoding of VC-1 Advanced profile via MediaCodec

### ISSUE THREAD

**Issue by CiNcH83** [CiNcH83/bravia_atv2#4]
The MediaTek MT5891 SoC is perfectly capable of decoding VC-1 Advanced as the stock _Video_ app shows no problem playing it back. This app however seems to have access to some private API definitions (via libstagefright).

Decoding VC-1 Advanced via the standard Android MediaCodec API is currently not properly supported and is therefore software decoded in Kodi via ffmpeg. The weak ARM cores of the MediaTek SoC obviously can’t handle the decoding of 1080p in real-time.

MediaTek drivers on Sony lack the respective codec:
```
OMX.MTK.VIDEO.DECODER.VC1                                video/wvc1
```

MediaTek drivers however expose:
```
OMX.MTK.VIDEO.DECODER.WMV                                video/x-ms-wmv
```

Some players instantiate the latter decoder in the hope that it can handle VC-1 Advanced profile. In VLC, this MTK decoder is blacklisted though as it only produces a black image for such content.

There is certainly a lack of information on how to properly handle VC-1 on MediaTek platforms.

([Here](http://www.blu-raystats.com/Stats/Stats.php?OrderBy=Title&VideoCodec=VC-1) is a list of over 700 Blu-ray titles (>15%) encoded in VC-1.)

**Comment by sjpotter** [comment]
it seems mx player knows how to play hardware accelerated vc-1.  tested last night and it shows its regular hw icon.  when one switches it to sw decoding, it becomes a frame dropping mess, like vlc is.

**Comment by CiNcH83** [comment]
Yes. Like the stock Video app, also MX Player can make use of libstagefright for decoding VC-1.

**Comment by MESWEB** [comment]
Hi. I have same problem. First movie contain VC-1 codec play very well and next episodes have 3-4 FPS like in your case. I'm using Default Video Player, VLC and MX Player. Only MX Player can play movie smoothly about 6 minutes and after FPS are reduced to minimum 3-4 FPS. For me this problem seems to be connected with RAM. Probably RAM are full and movie's slowing down. I have no idea what's the main reason of this problem. Maybe should be used custom codecs? Any tips?

**Comment by MESWEB** [comment]
Yesterday I finally finish watching full movie. I don't know how to do it. I was trying playing this video with several players MX Player, VLC and Video system app. Only stock Video app play video smoothly without problems. I make before factory reset of TV. But this hasn't help me and after 1h video play nice. I have BluRay disc copy - short trailers with same codec working very good in stock Video app but full movie length can't play as should be. I think probably some system apps are responsibility for this. I don't know which system app holds codecs and etc. I can force stop this app and restart and check if this help.

**Comment by MESWEB** [comment]
I do it again. Now I'm watching next episode with same codec. Step which one I have made:
SONY support write to me this e-mail:

> Thank you for your recent email.

>With reference to your enquiry, kindly test the app while the TV is connected to Hotspot.

>Connect your TV to hot-spot

>- Enable the mobile data on your mobile phone.

>- Enable the hot spot from your mobile phone.

>- Connect the TV to the Hot spot WIFI coming from your mobile phone following the below: (Home - help - settings - network - network setup - easy - wifi - connect by scan list - then choose the name of your smart-phone and enter the password).

>- Use the steps in the below link to connect your TV to the mobile Hotspot:

>https://helpguide.sony.net/tv/gaep1/v1/en/06-02_01.html

>- Try to use the application and see the outcome.

>- Disconnect the Hotspot from your mobile phone.

>Should you have any further questions, please feel free to reply to this e-mail.

>Thank you for your enquiry.

>Yours sincerely,


>Best regards,

>Esraa

>SONY SUPPORT TEAM

First restart TV and next I switch off Wi-Fi module in settings. Next step what I have done it was open file in MX Player and set up HW Decoding after 4 minutes when video start slowing down FPS I exit MX Player and open Video App built-in. I open trailer (2 minutes long - VC-1 codec) (please keep it mind all trailers VC-1 working good only full length movies have problem to play smooth) and I watched full this trailer and after trailer app exit to file list. So I try open full length movie size file and Voila. Success movie start working with full FPS. Maybe this Wi-Fi module causing problem - I don't know yet but anybody can try it.

**Comment by CiNcH83** [comment]
@MESWEB I am trying to keep this place as clean as possible so that it can act as a reference. Your postings are adding more noise, not really belonging to this issue but seemly being a mixture of other issues also mentioned in this tracker:

- [WLAN throughput often dropping to 1/4th of the available bandwidth](https://github.com/CiNcH83/bravia_atv2/issues/68)
- [Video app plays VC-1 within MKV with severe stuttering](https://github.com/CiNcH83/bravia_atv2/issues/76)
